packages/auth/amplify_auth_cognito/lib/amplify_auth_cognito.dart

```diff
@@ -16,28 +16,21 @@
 import 'dart:async';
 import 'dart:core';
 import 'package:amplify_auth_cognito/src/CognitoDevice/cognito_device.dart';
-import 'package:amplify_auth_cognito/src/CognitoHubEvents/AuthHubEvent.dart';
-import 'package:amplify_auth_plugin_interface/amplify_auth_plugin_interface.dart';
-import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:amplify_core/amplify_core.dart';
 import './method_channel_auth_cognito.dart';
 import './amplify_auth_cognito_stream_controller.dart';
 
 export './src/types.dart';
-export 'package:amplify_auth_plugin_interface/src/types.dart';
 
 class AmplifyAuthCognito extends AuthPluginInterface {
-  static final Object _token = Object();
-
   /// Constructs a AmplifyAuthCognito plugin
-  AmplifyAuthCognito() : super(token: _token);
+  AmplifyAuthCognito();
 
   static AmplifyAuthCognito _instance = AmplifyAuthCognitoMethodChannel();
   static AuthStreamController streamWrapper = AuthStreamController();
 
-  static set instance(AuthPluginInterface instance) {
```

Comment:
So we don't need this necessarily, but we do need a way to change where calls go depending on the platform. The instance + Dart's plugin registration system is one way, or you can have a factory method like this:

```dart
factory AmplifyAuthCognito() {
  if (zIsWeb || Platform.isMacOS || Platform.isWindows || Platform.isLinux) {
    return AmplifyAuthCognitoDart();
  } else {
    return AmplifyAuthCognitoMethodChannel();
  }
}
```

and then you can make this class abstract and remove all methods.

Reply:
Yup - will bring this in once we actually have the Dart implementation.